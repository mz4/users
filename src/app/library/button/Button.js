import PropTypes from 'prop-types';
import styles from './Button.module.scss';

export const Button = ({ actionButton, text, type, dataTestId }) => {
  const buttonType = {
    one: styles.one,
    two: styles.two,
    three: styles.three,
    four: styles.four,
    five: styles.five,
  };

  return (
    <button
      onClick={actionButton}
      className={buttonType[type]}
      data-testid={dataTestId}
    >
      {text}
    </button>
  );
};

Button.propTypes = {
  actionButton: PropTypes.func.isRequired,
  text: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  dataTestId: PropTypes.string.isRequired,
};
