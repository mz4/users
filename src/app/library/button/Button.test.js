import renderer from 'react-test-renderer';
import { fireEvent, render, screen } from '@testing-library/react';
import { Button } from './Button';

describe('It renders a button component', () => {
  const handleClick = jest.fn();
  const btn = (
    <Button actionButton={handleClick} type="secondary" text="submit" />
  );
  it('Create a snapshot of the button', () => {
    const component = renderer.create(btn);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Render button with submit text', () => {
    const { getByText, getByRole, container } = render(btn);
    const element = container.getElementsByClassName('secondary');
    expect(element.length).toBe(1);
    expect(getByText('submit')).toBeTruthy();
    expect(getByRole('button')).toBeEnabled();
  });
  it('Call a function onClick event', () => {
    render(btn);
    fireEvent.click(screen.getByText('submit'));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
