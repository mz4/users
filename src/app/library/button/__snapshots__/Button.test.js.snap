// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders a button component Create a snapshot of the button 1`] = `
<button
  className="secondary"
  onClick={[MockFunction]}
>
  submit
</button>
`;
